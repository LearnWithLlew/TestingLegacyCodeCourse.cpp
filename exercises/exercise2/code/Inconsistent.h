#pragma once
#include <cstdlib>
#include <ctime>
#include <string>
#include <iostream>
#include <sstream>
#include <ctime>
#include <chrono>

class Inconsistent
{
public:
    int getID()
    {
        srand(time(0));
        return (rand() % 10 + 1);
    }

    std::string print(const std::string& name)
    {
        std::stringstream stream;
        std::chrono::system_clock::time_point end = std::chrono::system_clock::now();
        std::time_t end_time =
                std::chrono::system_clock::to_time_t(end);
        auto time = std::ctime(&end_time);
        time[strcspn(time, "\n")] = '\0';
        stream << "started at " << time << " by " << name;
        return stream.str();
    }

    std::string print1(const std::string& name)
    {
        auto end = std::chrono::system_clock::now();
        return print2(name, end);
    }

    std::string print2(const std::string &name, std::chrono::system_clock::time_point end) const {
        std::stringstream stream;
        time_t end_time =
                std::chrono::system_clock::to_time_t(end);
        auto time = std::ctime(&end_time);
        time[strcspn(time, "\n")] = '\0';
        stream << "started at " << time << " by " << name;
        return stream.str();
    }
};
