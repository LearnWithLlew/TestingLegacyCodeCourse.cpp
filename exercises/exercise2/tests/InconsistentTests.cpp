#include <catch2/catch.hpp>
#include "ApprovalTests.hpp"
//#include "Samples.h"

#include "Inconsistent.h"

using namespace ApprovalTests;

TEST_CASE("Poke GetID")
{
}

TEST_CASE("Poke print")
{
    Inconsistent inconsistent;
    Approvals::verify(inconsistent.print2("fred", DateUtils::createDateTime(2000, 01, 02, 3, 4, 5)));
}

TEST_CASE("Scrub print")
{
    Inconsistent inconsistent;
    Approvals::verify(inconsistent.print1("fred"), Options().withScrubber(Scrubbers::createRegexScrubber("fixme", "[date]")));
}
