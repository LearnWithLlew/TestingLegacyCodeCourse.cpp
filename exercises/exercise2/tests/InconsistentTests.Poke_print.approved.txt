started at Sun Jan  2 03:04:05 2000 by fred
